lib/Serialization/ASTWriter.cpp

```diff
@@ -334,6 +334,13 @@ void ASTTypeWriter::VisitTypedefType(const TypedefType *T) {
   Code = TYPE_TYPEDEF;
 }
 
+void ASTTypeWriter::VisitTypeVariableType(const TypeVariableType *T) {
```

Comment:
There should be matching code in ASTReader.cpp for reading the type variable in.    I think the right function for implementing the functionality is readTypeRecord.